packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -378,6 +384,9 @@ class _ModalScope extends StatefulWidget {
 }
 
 class _ModalScopeState extends State<_ModalScope> {
+  // See addScopedWillPopCallback, removeScopedWillPopCallback in ModalRoute.
+  List<WillPopCallback> willPopCallbacks = <WillPopCallback>[];
```

Comment:
I had it that way originally. Doing so makes all of the other references to willPopCallbacks messier; I don't think there's much of a win here.